src/java/main/org/apache/zookeeper/server/ZooKeeperServerMain.java

```diff
@@ -146,14 +146,26 @@ public void runFromConfig(ServerConfig config) throws IOException, AdminServerEx
             );
             containerManager.start();
 
+            // Watch status of ZooKeeper server. If there is an internal error
+            // then will do a graceful shutdown.
+            while (zkServer.isRunning()) {
```

Comment:
If someone needs a different usage in standalone server, they can do a different impl of this interface and pass it through ZKServer constructor rather than extending the impl class. I thought exposing an interface would be more cleaner.